backend/sirius-web-emf/src/main/java/org/eclipse/sirius/web/emf/view/ViewPropertiesConfigurer.java

```diff
@@ -0,0 +1,171 @@
+                              .labelProvider(variableManager -> "Properties") //$NON-NLS-1$
+                              .semanticElementsProvider(this.semanticElementsProvider)
+                              .canCreatePredicate(canCreatePredicate)
+                              .groupDescriptions(List.of(groupDescription))
+                              .build();
+        // @formatter:on
+    }
+
+    private GroupDescription createSimpleGroupDescription(List<AbstractControlDescription> controls) {
+        // @formatter:off
+        return GroupDescription.newGroupDescription("group") //$NON-NLS-1$
+                               .idProvider(variableManager -> "group") //$NON-NLS-1$
+                               .labelProvider(variableManager -> "General") //$NON-NLS-1$
+                               .semanticElementsProvider(this.semanticElementsProvider)
+                               .controlDescriptions(controls)
+                               .build();
+        // @formatter:on
+    }
+
+    private TextfieldDescription createTextField(String id, String title, Function<Object, String> reader, BiConsumer<Object, String> writer) {
+        Function<VariableManager, String> valueProvider = variableManager -> variableManager.get(VariableManager.SELF, Object.class).map(reader).orElse(EMPTY);
+        BiFunction<VariableManager, String, Status> newValueHandler = (variableManager, newValue) -> {
+            var optionalDiagramMapping = variableManager.get(VariableManager.SELF, Object.class);
+            if (optionalDiagramMapping.isPresent()) {
+                writer.accept(optionalDiagramMapping.get(), newValue);
+                return Status.OK;
+            } else {
+                return Status.ERROR;
+            }
+        };
+        // @formatter:off
+        return TextfieldDescription.newTextfieldDescription(id)
+                                   .idProvider(variableManager -> id)
+                                   .labelProvider(variableManager -> title)
+                                   .valueProvider(valueProvider)
+                                   .newValueHandler(newValueHandler)
+                                   .build();
+        // @formatter:on
+    }
+
+    private SelectDescription createShapeSelectionField() {
+        // @formatter:off
+        return SelectDescription.newSelectDescription("nodestyle.shapeSelector") //$NON-NLS-1$
+                                .idProvider(variableManager -> "nodestyle.shapeSelector") //$NON-NLS-1$
+                                .labelProvider(variableManager -> "Shape") //$NON-NLS-1$
+                                .valueProvider(variableManager -> variableManager.get(VariableManager.SELF, NodeStyle.class).map(NodeStyle::getShape).orElse(EMPTY))
+                                .optionsProvider(variableManager -> this.customImagesService.getAvailableImages().stream().collect(Collectors.toList()))
```

Comment:
The following call returns a list : `this.customImagesService.getAvailableImages()` so `.stream().collect(Collectors.toList())` is not useful unless I missed something

Reply:
I get a `Type mismatch: cannot convert from List<CustomImage> to List<Object>` without the collect. Anyway, I've modified this line a little since to return the images sorted by label, so the intermediate step is needed anyway.